fakes/image.go

```diff
@@ -37,32 +43,276 @@ func NewImage(name, topLayerSha string, identifier imgutil.Identifier) *Image {
 	}
 }
 
+var ErrLayerNotFound = errors.New("layer with given diff id not found")
+
 type Image struct {
-	deleted          bool
-	layers           []string
-	history          []v1.History
-	layersMap        map[string]string
-	prevLayersMap    map[string]string
-	reusedLayers     []string
-	labels           map[string]string
-	env              map[string]string
-	topLayerSha      string
-	os               string
-	osVersion        string
-	architecture     string
-	variant          string
-	identifier       imgutil.Identifier
-	name             string
-	entryPoint       []string
-	cmd              []string
-	base             string
-	createdAt        time.Time
-	layerDir         string
-	workingDir       string
-	savedNames       map[string]bool
-	manifestSize     int64
-	refName          string
-	savedAnnotations map[string]string
+	deleted                    bool
+	layers                     []string
+	history                    []v1.History
+	layersMap                  map[string]string
+	prevLayersMap              map[string]string
+	reusedLayers               []string
+	labels                     map[string]string
+	env                        map[string]string
+	topLayerSha                string
+	os                         string
+	osVersion                  string
+	architecture               string
+	variant                    string
+	identifier                 imgutil.Identifier
+	name                       string
+	entryPoint                 []string
+	cmd                        []string
+	base                       string
+	createdAt                  time.Time
+	layerDir                   string
+	workingDir                 string
+	savedNames                 map[string]bool
+	manifestSize               int64
+	refName                    string
+	savedAnnotations           map[string]string
+	features, osFeatures, urls []string
+}
+
+func mapToStringSlice(data map[string]string) []string {
+	var stringSlice []string
+	for key, value := range data {
+		keyValue := fmt.Sprintf("%s=%s", key, value)
+		stringSlice = append(stringSlice, keyValue)
+	}
+	return stringSlice
+}
+
+// ConfigFile implements v1.Image.
+func (i *Image) ConfigFile() (*v1.ConfigFile, error) {
```

Comment:
This doesn't look like a **Fake** implementation, looks like we want to duplicate the real logic and it doesn't make a lot of sense.